src/utils.re

```diff
@@ -0,0 +1,193 @@
+          } else if (
+            accum.[String.length accum - 1] == '-'
+          ) {
+            String.slice accum 0 (-1) ^ (Char.to_string char |> String.capitalize)
+          } else {
+            accum ^ Char.to_string char
+          }
+      );
+  let bsLibToModule (Lib name) => Mod (String.capitalize name |> bsKebabToCamel);
+  tsm (bsLibToModule libName) ^ "__" ^ tsm (pathToModule path)
+};
+
+/* let topLibName = {
+     let packageJsonPath = Path.relative dir::Path.the_root "package.json";
+     from_file (Path.to_string packageJsonPath) |> Util.member "name" |> Util.to_string |> (
+       fun name => Lib name
+     )
+   }; */
+let topLibName = Lib "src";
+
+/* Generic sorting algorithm on directed acyclic graph. Example: [(a, [b, c, d]), (b, [c]), (d, [c])] will be
+   sorted into [c, d, b, a] or [c, b, d, a], aka the ones being depended on will always come before the
+   dependent */
+let topologicalSort graph => {
+  let graph = {contents: graph};
+  let rec topologicalSort' currNode accum => {
+    let nodeDeps =
+      switch (List.Assoc.find graph.contents currNode) {
+      /* node not found: presume to be third-party dep. This is slightly dangerous because it might also mean
+         we didn't construct the graph correctly. */
+      | None => []
+      | Some nodeDeps' => nodeDeps'
+      };
+    List.iter nodeDeps f::(fun dep => topologicalSort' dep accum);
+    if (List.for_all accum.contents f::(fun n => n != currNode)) {
+      accum := [currNode, ...accum.contents];
+      graph := List.Assoc.remove graph.contents currNode
+    }
+  };
+  let accum = {contents: []};
+  while (not (List.is_empty graph.contents)) {
+    topologicalSort' (fst (List.hd_exn graph.contents)) accum
+  };
+  List.rev accum.contents
+};
+
+/* package.json helpers */
+let targets = {
+  let packageJsonPath = Path.relative dir::Path.the_root "package.json";
+  let targets' =
+    from_file (Path.to_string packageJsonPath) |> Util.member "rebel" |>
+    Util.to_option (fun a => a |> Util.member "targets");
+  switch targets' {
+  | Some (`List _ as a) => List.map f::Util.to_string (Util.to_list a)
+  | None => []
+  | _ => []
+  }
+};
+
+let backend = {
+  let packageJsonPath = Path.relative dir::Path.the_root "package.json";
+  let optBackend =
+    from_file (Path.to_string packageJsonPath) |> Util.member "rebel" |>
+    Util.to_option (fun a => a |> Util.member "backend");
+  switch optBackend {
+  | Some (`String _ as a) =>
+    let backend' = Util.to_string a;
+    List.mem ["native", "jsoo", "bucklescript"] backend' ? backend' : "native"
+  | _ => "native"
+  }
+};
+
+let bscFlags = {
+  let packageJsonPath = Path.relative dir::Path.the_root "package.json";
+  let flags =
+    from_file (Path.to_string packageJsonPath) |> Util.member "rebel" |>
+    Util.to_option (fun a => a |> Util.member "bscFlags");
+  switch flags {
+  | Some (`String _ as a) => Util.to_string a
+  | _ => ""
+  }
+};
+
+let ocamlcFlags = {
```

Comment:
@chenglou  This can handle #12 I didn't completely flesh it. Since there is still some time before this gets merged. Let me complete this also.